### PR TITLE
Fixed #1227 - 2.0: Authentication (Replicator)

### LINKED ISSUE #1227: 2.0: Authentication (Replicator)

### PR DESCRIPTION
**Pull request by hideki**
- Update Replicator APIs to follow latest spec.

### CODE REVIEW

File: shared/src/main/java/com/couchbase/lite/Authenticator.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.Map;
+
+/**
+ * Authenticator is an opaque based authenticator interface and not intended for application to
+ * implement a custom authenticator by subclassing Authenticator interface.
+ */
+public abstract class Authenticator {
+    static final String kCBLReplicatorAuthOption = "auth";
+    static final String kCBLReplicatorAuthUserName = "username";
+    static final String kCBLReplicatorAuthPassword = "password";
```
Comment (pasin) [inline shared/src/main/java/com/couchbase/lite/Authenticator.java]: I'm not sure if these belong here. Shouldn't there be at the C4Replicator?

File: shared/src/main/java/com/couchbase/lite/SessionAuthenticator.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public String getCookieName() {
+        return cookieName;
+    }
+
+    //---------------------------------------------
+    // Authenticator abstract method implementation
+    //---------------------------------------------
+    @Override
+    void authenticate(Map<String, Object> options) {
+
```
Comment (pasin) [inline shared/src/main/java/com/couchbase/lite/SessionAuthenticator.java]: Missing implementation.

File: shared/src/main/java/com/couchbase/lite/Authenticator.java (hunk 1 above)
Comment (hideki) [inline shared/src/main/java/com/couchbase/lite/Authenticator.java]: Currently `C4Socket` is implemented in `litecore` layer. It should be in `lite` layer. I'd like to keep these constants in Autheticator till `C4Socket` is moved to `lite` layer.

File: shared/src/main/java/com/couchbase/lite/SessionAuthenticator.java (hunk 2 above)
Comment (hideki) [inline shared/src/main/java/com/couchbase/lite/SessionAuthenticator.java]: I forgot to implement this...

**Comment by hideki** [comment]
Implemented `SessionAuthenticator.authenticate()`. I'd like to keep constants in Authenticator till C4Socket is refactored. Thanks!

**Pull request merged**